src/v/cloud_storage_clients/abs_client.h

```diff
@@ -286,6 +286,13 @@ class abs_client : public client {
     ss::future<storage_account_info>
     do_get_account_info(ss::lowres_clock::duration timeout);
 
+    /// \brief test if Hierarchical Namespace is enabled in the storage account
+    /// by creating a test file and calling Set Expiry on this. Set Expiry is
+    /// available only when HNS is enabled. use the result to infer if HNS is
+    /// enabled.
```

Comment:
is this a strategy you've seen used elsewhere, or documented somewhere? where can i learn more?

Reply:
updated the cover letter with a couple of links and I'll add details in the confluence page for this feature